### PR TITLE
feat(baremetal): add private networks

### PR DESCRIPTION
**Pull request by yfodil**

### CODE REVIEW

File: docs/resources/baremetal_server.md [hunk 1]
```diff
@@ -107,6 +153,14 @@ In addition to all above arguments, the following attributes are exported:
 - `id` - The ID of the server.
 - `offer_id` - The ID of the offer.
 - `os_id` - The ID of the os.
+- `private_network` - The private networks attached to the server.
+  -> The `private_network` block contains :
```
Comment (Monitob) [inline docs/resources/baremetal_server.md:157]:
```suggestion
```

File: scaleway/resource_baremetal_server.go [hunk 2]
```diff
@@ -305,15 +377,30 @@ func resourceScalewayBaremetalServerRead(ctx context.Context, d *schema.Resource
...
 	_ = d.Set("options", flattenBaremetalOptions(server.Zone, server.Options))
 
+	listPrivateNetworks, err := baremetalPrivateNetworkAPI.ListServerPrivateNetworks(&baremetal.PrivateNetworkAPIListServerPrivateNetworksRequest{
+		Zone:     server.Zone,
+		ServerID: &server.ID,
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to list server's private networks: %w", err))
+	}
+	_ = d.Set("private_network", flattenBaremetalPrivateNetworks(server.Zone, listPrivateNetworks.ServerPrivateNetworks))
```
Comment (Monitob) [inline scaleway/resource_baremetal_server.go:387]: Why the zone is not returned on the `ServerPrivateNetworks`? This should be added

**Review by Monitob — CHANGES_REQUESTED** [review]

File: scaleway/resource_baremetal_server.go (hunk 2 above)
Comment (yfodil) [reply scaleway/resource_baremetal_server.go:409]: It's returned in the id

**Review by remyleone — DISMISSED** [review]

File: scaleway/helpers_baremetal.go [hunk 3]
```diff
@@ -129,11 +169,26 @@ func flattenBaremetalOptions(zone scw.Zone, options []*baremetal.ServerOption) i
...
 	}
 	return flattenedOptions
 }
 
+func flattenBaremetalPrivateNetworks(zone scw.Zone, privateNetworks []*baremetal.ServerPrivateNetwork) interface{} {
+	flattenedPrivateNetworks := []map[string]interface{}(nil)
+	for _, privateNetwork := range privateNetworks {
+		flattenedPrivateNetworks = append(flattenedPrivateNetworks, map[string]interface{}{
+			"id":         newZonedID(zone, privateNetwork.PrivateNetworkID).String(),
+			"vlan":       privateNetwork.Vlan,
```
Comment (Monitob) [inline scaleway/helpers_baremetal.go:183]: Please note that this is a pointer. verify if null

**Review by Monitob — CHANGES_REQUESTED** [review]

**Review by Monitob — APPROVED** [review]

**Pull request merged**